crd/overlayextensionconfig/api/v1alpha1/overlayextensionconfig_types.go

```diff
@@ -0,0 +1,57 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OverlayExtensionConfig is the Schema for the overlayextensionconfigs API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=oec
+// +kubebuilder:printcolumn:name="OverlayExtensionConfig IP range",type=string,priority=1,JSONPath=`.spec.extensionIPRange`
+type OverlayExtensionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OverlayExtensionConfigSpec   `json:"spec,omitempty"`
+	Status OverlayExtensionConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// OverlayExtensionConfigList contains a list of OverlayExtensionConfig
+type OverlayExtensionConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OverlayExtensionConfig `json:"items"`
+}
+
+// OverlayExtensionConfigSpec defines the desired state of OverlayExtensionConfig
+type OverlayExtensionConfigSpec struct {
+	// +kubebuilder:validation:Optional
```

Comment:
is there a way to comment here so that description in the generated manifest gets updated?

Reply:
yes Use this 

type ToySpec struct {
    // +kubebuilder:validation:MaxLength=15
    // +kubebuilder:validation:MinLength=1
    // +kubebuilder:validation:Description="Name of the toy"
    Name string `json:"name,omitempty"`

    // +kubebuilder:validation:MaxItems=500
    // +kubebuilder:validation:MinItems=1
    // +kubebuilder:validation:UniqueItems=true
    // +kubebuilder:validation:Description="List of knights"
    Knights []string `json:"knights,omitempty"`
}

Reply:
this was just me asking copilot "when using controller gen to generate a crd from code how do you populate descriptions

Reply:
No, it doesn't look like ChapGTP is correct this time. This marker doesn't seem to exist: 
`// +kubebuilder:validation:Description`
link to documentation: https://book.kubebuilder.io/reference/markers.html

I put a better comment above `type OverlayExtensionConfigSpec`, which is copied to `.spec.description` firld in the CRD.